eradio.el

```diff
@@ -46,13 +46,29 @@ This is a list of the program and its arguments.  The url will be appended to th
 
 (defvar eradio-process nil "The process running the radio player.")
 
+(defvar eradio-current-channel nil "Currently playing (or paused) channel.")
+
 (defun eradio-alist-keys (alist)
   "Get the keys from an ALIST."
   (mapcar #'car alist))
 
 (defun eradio-stop ()
   "Stop the radio player."
-  (interactive) (when eradio-process (delete-process eradio-process)))
+  (interactive)
+  (when eradio-process
+    (delete-process eradio-process)
+    (setq eradio-current-channel nil)
+    (setq eradio-process nil)))
+
+(defun eradio-toggle ()
+  "Toggle the radio player."
+  (interactive)
+  (cond
+   (eradio-process
+    (delete-process eradio-process)
+    (setq eradio-process nil))
```

Comment:
Maybe this can be extracted as `eradio-pause` function, but I didn't want to introduce a new function without asking you first. What do you think?